sonic_db_config/db_config.go

```diff
@@ -1,135 +1,237 @@
+	if !sonic_db_init {
+		DbInit()
+	}
+	db_list := GetDbList(ns)
+	separator, ok := db_list[db_name].(map[string]interface{})["separator"]
+	if !ok {
+		panic(fmt.Errorf("'separator' is not a valid field in database_config.json file!"))
+	}
+	return separator.(string)
+}
+
+func GetDbId(db_name string, ns string) int {
+	if !sonic_db_init {
+		DbInit()
+	}
+	db_list := GetDbList(ns)
+	id, ok := db_list[db_name].(map[string]interface{})["id"]
+	if !ok {
+		panic(fmt.Errorf("'id' is not a valid field in database_config.json file!"))
+	}
+	return int(id.(float64))
+}
+
+func GetDbSock(db_name string, ns string) string {
+	if !sonic_db_init {
+		DbInit()
+	}
+	inst := GetDbInst(db_name, ns)
+	unix_socket_path, ok := inst["unix_socket_path"]
+	if !ok {
+		panic(fmt.Errorf("'unix_socket_path' is not a valid field in database_config.json file!"))
+	}
+	return unix_socket_path.(string)
+}
+
+func GetDbHostName(db_name string, ns string) string {
+	if !sonic_db_init {
+		DbInit()
+	}
+	inst := GetDbInst(db_name, ns)
+	hostname, ok := inst["hostname"]
+	if !ok {
+		panic(fmt.Errorf("'hostname' is not a valid field in database_config.json file!"))
+	}
+	return hostname.(string)
+}
+
+func GetDbPort(db_name string, ns string) int {
+	if !sonic_db_init {
+		DbInit()
+	}
+	inst := GetDbInst(db_name, ns)
+	port, ok := inst["port"]
+	if !ok {
+		panic(fmt.Errorf("'port' is not a valid field in database_config.json file!"))
+	}
+	return int(port.(float64))
+}
+
+func GetDbTcpAddr(db_name string, ns string) string {
+	if !sonic_db_init {
+		DbInit()
+	}
+	hostname := GetDbHostName(db_name, ns)
+	port := GetDbPort(db_name, ns)
+	return hostname + ":" + strconv.Itoa(port)
+}
+
+func DbGetNamespaceAndConfigFile(ns_to_cfgfile_map map[string]string) {
+	data, err := io.ReadFile(SONIC_DB_GLOBAL_CONFIG_FILE)
+	if err == nil {
+		//Ref:https://stackoverflow.com/questions/18537257/how-to-get-the-directory-of-the-currently-running-file
+		dir, err := filepath.Abs(filepath.Dir(SONIC_DB_GLOBAL_CONFIG_FILE))
+		if err != nil {
+			panic(err)
+		}
+		var sonic_db_global_config = make(map[string]interface{})
+		err = json.Unmarshal([]byte(data), &sonic_db_global_config)
+		if err != nil {
+			panic(err)
+		}
+		var db_include_file string
+		for _, entry := range sonic_db_global_config["INCLUDES"].([]interface{}) {
+			ns, ok := entry.(map[string]interface{})["namespace"]
+			if !ok {
+				ns = SONIC_DEFAULT_NAMESPACE
+			}
+			_, ok = ns_to_cfgfile_map[ns.(string)]
+			if ok {
+				panic(fmt.Errorf("Global Database config file is not valid(multiple include for same namespace!"))
+			}
+			//Ref:https://www.geeksforgeeks.org/filepath-join-function-in-golang-with-examples/
+			db_include_file = filepath.Join(dir, entry.(map[string]interface{})["include"].(string))
+			ns_to_cfgfile_map[ns.(string)] = db_include_file
+		}
+		if len(ns_to_cfgfile_map) > 1 {
+			sonic_db_multi_namespace = true
+		} else {
+			sonic_db_multi_namespace = false
```

Comment:
yes in that case it will be same as regular case where there is only one redis instance running in host. It is not an error case.
These files are static are built into sonic image based on platform type so can't/shouldn't be change run-time.